### DISCUSSION TITLE
Single-file source code

### DISCUSSION THREAD

**Discussion by ghost** [nxrighthere/UnrealCLR#230] [General]
Is there any other reason why the different types are not in different files other than a matter of taste?

**Comment by nxrighthere** [comment]
Nope, it's just my personal preference.

**Comment by ghost** [comment]
It would be nice to extract the classes. It is easier to work with individual classes when the project gets bigger and more people get involved to contribute. Scrolling is annoying. Merging big files too. I can do that if there is a consensus on this issue.

I have additional question about the Codegen.cs. There are comment "Automatically generated" above the Shared class. How is generated this class or file?

**Comment by nxrighthere** [comment]
> Scrolling is annoying.

For me, it's annoying to maintain and navigate through a thousand files in a project. The structure is optimized for my own convenience since 99,9% of the code is written and maintained by me.

> I have additional question about the Codegen.cs. There are comment "Automatically generated" above the Shared class. How is generated this class or file?

Please, see https://github.com/nxrighthere/UnrealCLR/discussions/11.
